lib/HdtDocument.cc

```diff
@@ -65,6 +65,124 @@ const Nan::Persistent<Function>& HdtDocument::GetConstructor() {
 }
 
 
+/******** Generate HDT doc from rdf file ********/
+
+class Rdf2HdtWorker : public Nan::AsyncProgressWorker {
+  string fromFile;
+  string toFile;
+  string rdfFormat;
+  string baseUri;
+  Nan::Callback *progressCallback;
+  HDTSpecification options;
+  HDT* hdt;
+
+public:
+  class NodeProgressListener : public ProgressListener {
+    private:
+    public:
+      double lastLevel = -1.0;
+      const AsyncProgressWorker::ExecutionProgress* nanExecutionProgress;
+      NodeProgressListener(const AsyncProgressWorker::ExecutionProgress* nanExecutionProgress) {this->nanExecutionProgress = nanExecutionProgress;}
+      virtual ~NodeProgressListener() {}
+
+      void sendToNode(float level) {
+        double dlevel = level;
+        if (dlevel == lastLevel || dlevel == 100.0) {
```

Comment:
Dlevel is already cast to double, so in this case we're comparing doubles instead of floats. But, I agree that adding a threshold is better. What about rounding the floats to int (and returning js integers in the callback instead of js numbers). That should give us an intuitive threshold